packages/cli/src/controller/generate-controller.ts

```diff
@@ -0,0 +1,317 @@
+    return inputString.slice(removeString.length);
+  }
+  return inputString;
+}
+
+export function constructMethod<T extends ConstructorFragment | Fragment>(
+  cleanedFragment: Record<string, T>
+): SelectedMethod[] {
+  return Object.keys(cleanedFragment).map((f) => {
+    return {
+      name: cleanedFragment[f].name,
+      method: f,
+    };
+  });
+}
+
+export async function promptSelectables<T extends ConstructorFragment | Fragment>(
+  method: 'event' | 'function',
+  availableMethods: Record<string, T>
+): Promise<Record<string, T>> {
+  const selectedMethods: Record<string, T> = {};
+  const chosenFn = await inquirer.prompt({
+    name: method,
+    message: `Select ${method}`,
+    type: 'checkbox',
+    choices: Object.keys(availableMethods),
+  });
+  const choseArray = chosenFn[method] as string[];
+  choseArray.map((choice: string) => {
+    selectedMethods[choice] = availableMethods[choice];
+  });
+
+  return selectedMethods;
+}
+
+export async function renderTemplate(templatePath: string, outputPath: string, templateData: ejs.Data): Promise<void> {
+  const data = await ejs.renderFile(templatePath, templateData);
+  await fs.promises.writeFile(outputPath, data);
+}
+
+export function getAbiInterface(projectPath: string, abiPath: string): Interface {
+  const abi = loadFromJsonOrYaml(path.join(projectPath, abiPath)) as any;
+  return new Interface(abi);
+}
+
+export function filterObjectsByStateMutability(
+  obj: Record<string, FunctionFragment>
+): Record<string, FunctionFragment> {
+  const filteredObject: Record<string, FunctionFragment> = {};
+  for (const key in obj) {
+    if (obj[key].stateMutability !== 'view') {
+      filteredObject[key] = obj[key];
+    }
+  }
+  return filteredObject;
+}
+
+export function getFragmentFormats<T extends ConstructorFragment | Fragment>(fragment: T): {full: string; min: string} {
+  return {
+    full: removeKeyword(fragment.format('full')),
+    min: removeKeyword(fragment.format('minimal')),
+  };
+}
+
+export function generateHandlerName(name: string, abiName: string, type: 'tx' | 'log'): string {
+  return `handle${upperFirst(name)}${upperFirst(abiName)}${upperFirst(type)}`;
+}
+
+export function constructDatasources(userInput: UserInput): EthereumDs {
+  const abiName = parseContractPath(userInput.abiPath).name;
+  const formattedHandlers: SubqlRuntimeHandler[] = [];
+
+  userInput.functions.map((fn) => {
+    const handler: SubqlRuntimeHandler = {
+      handler: generateHandlerName(fn.name, abiName, 'tx'),
+      kind: EthereumHandlerKind.Call,
+      filter: {
+        function: fn.method,
+      },
+    };
+    formattedHandlers.push(handler);
+  });
+
+  userInput.events.map((event) => {
+    const handler: SubqlRuntimeHandler = {
+      handler: generateHandlerName(event.name, abiName, 'log'),
+      kind: EthereumHandlerKind.Event,
+      filter: {
+        topics: [event.method],
+      },
+    };
+    formattedHandlers.push(handler);
+  });
```

Comment:
Map should return a value. You could use `.forEach` here if you don't want to do that instead